File: src/main/java/cn/don9cn/blog/action/bussiness/subscribe/SubscribeAction.java
package cn.don9cn.blog.action.bussiness.subscribe;

import cn.don9cn.blog.action.BaseAction;
import cn.don9cn.blog.model.bussiness.subscribe.SubscribeInfo;
import cn.don9cn.blog.plugins.daohelper.core.PageResult;
import cn.don9cn.blog.service.bussiness.subscribe.interf.SubscribeService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;

/**
 * @author Don9
 * @create 2017-12-27-13:18
 **/
@RestController
@RequestMapping(value = "/bussiness/subscribe")
public class SubscribeAction extends BaseAction<SubscribeInfo>{

    @Autowired
    private SubscribeService subscribeService;

    @Override
    @PostMapping("/public")
    protected Object baseInsert(SubscribeInfo subscribeInfo) {
        return subscribeService.baseInsert(subscribeInfo);
    }

    @Override
    protected Object baseInsertBatch(List<SubscribeInfo> list) {
        return null;
    }

    @Override
    protected Object baseUpdate(SubscribeInfo subscribeInfo) {
        return null;
    }

    @Override
    protected Object baseRemove(String code) {
        return null;
    }

    @Override
    protected Object baseRemoveBatch(String codes) {
        return null;
    }

    @Override
    protected Object baseFindById(String code) {
        return null;
    }

    @Override
    protected Object baseFindAll() {
        return null;
    }

    @Override
    @GetMapping("/list")
    protected Object baseFindListByParams(SubscribeInfo subscribeInfo) {
        return subscribeService.baseFindListByParams(subscribeInfo);
    }

    @Override
    @GetMapping("/page")
    protected Object baseFindByPage(int page, int limit, String startTime, String endTime, String orderBy, SubscribeInfo subscribeInfo) {
        return subscribeService.baseFindByPage(new PageResult<>(page,limit,startTime,endTime,orderBy,subscribeInfo));
    }

    @DeleteMapping("/public")
    public Object delete(String email,String author){
        return subscribeService.delete(email, author);
    }



}

File: src/main/java/cn/don9cn/blog/action/bussiness/subscribe/SubscribeAction.kt
package cn.don9cn.blog.action.bussiness.subscribe

import cn.don9cn.blog.support.mongo.core.PageResult
import cn.don9cn.blog.action.BaseAction
import cn.don9cn.blog.model.bussiness.subscribe.SubscribeInfo
import cn.don9cn.blog.service.bussiness.SubscribeService
import cn.don9cn.blog.support.action.ActionMsg
import cn.don9cn.blog.support.action.ActionMsgHandler
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping(value = ["/bussiness/subscribe"])
open class SubscribeAction : BaseAction<SubscribeInfo>() {

    @Autowired
    private var subscribeService: SubscribeService? = null

    @PostMapping("/public")
    override fun baseInsert(t: SubscribeInfo): ActionMsg {
        return subscribeService!!.insert(t)
    }

    override fun baseInsertBatch(list: List<SubscribeInfo>): ActionMsg {
        return ActionMsgHandler.insert(subscribeService!!.baseInsertBatch(list))
    }

    override fun baseUpdate(t: SubscribeInfo): ActionMsg {
        return ActionMsgHandler.update(subscribeService!!.baseUpdate(t))
    }


    override fun baseRemove(code: String): ActionMsg {
        return ActionMsgHandler.delete(subscribeService!!.baseDeleteById(code))
    }

    @DeleteMapping
    override fun baseRemoveBatch(codes: String): ActionMsg {
        return ActionMsgHandler.delete(subscribeService!!.baseDeleteBatch(codes))
    }

    override fun baseFindById(code: String): ActionMsg {
        return ActionMsgHandler.find(subscribeService!!.baseFindById(code))
    }

    override fun baseFindAll(): ActionMsg {
        return ActionMsgHandler.find(subscribeService!!.baseFindAll())
    }

    override fun baseFindListByParams(t: SubscribeInfo): ActionMsg {
        return ActionMsgHandler.find(subscribeService!!.baseFindListByParams(t))
    }

    @GetMapping("/page")
    override fun baseFindByPage(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, t: SubscribeInfo): ActionMsg {
        return ActionMsgHandler.find(subscribeService!!.baseFindByPage(PageResult(page, limit, startTime, endTime, orderBy, t)))
    }

    @GetMapping("/page/byUser")
    open fun findByPageByUser(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, t: SubscribeInfo): ActionMsg {
        return ActionMsgHandler.find(subscribeService!!.findByPageByUser(PageResult(page, limit, startTime, endTime, orderBy, t)))
    }

    @GetMapping("/subscribes/page/byUser")
    open fun findSubscribesByPageByUser(page: Int, limit: Int, startTime: String?, endTime: String?, orderBy: String?, t: SubscribeInfo): ActionMsg {
        return ActionMsgHandler.find(subscribeService!!.findSubscribesByPageByUser(PageResult(page, limit, startTime, endTime, orderBy, t)))
    }

    @DeleteMapping("/public")
    open fun delete(email: String, author: String): ActionMsg {
        return subscribeService!!.delete(email, author)
    }

}
